src/Bicep.Decompiler/UniqueNamingResolver.cs

```diff
@@ -18,6 +22,9 @@ public class UniqueNamingResolver : INamingResolver
 
         private readonly Dictionary<string, string> assignedResourceNames = new(StringComparer.OrdinalIgnoreCase);
 
+        private readonly Regex ResourceNameRemoveTrailingNameRegex = new Regex("([a-zA-Z][a-zA-Z0-9_]*)Name([0-9]*)$");
```

Comment:
Should we ensure the prefix is non-zero length (`+` instead of `*`)? Just wondering what would happen if trying to rename a var named `name`.

Reply:
It's non-zero because of the initial [a-zA-Z] (technically names can start with "_" but I'd have to end up with just "_" for "_name").  I also have a unit test for "Name".